plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -0,0 +1,565 @@
+            .put("ShortTimestampType", SnowflakeClient::snowFlakeTimestampWriter)
+            .put("ShortTimestampWithTimeZoneType", SnowflakeClient::snowFlakeTimestampWithTZWriter)
+            .put("LongTimestampType", SnowflakeClient::snowFlakeTimestampWithTZWriter)
+            .put("LongTimestampWithTimeZoneType", SnowflakeClient::snowFlakeTimestampWithTZWriter)
+            .put("VarcharType", SnowflakeClient::snowFlakeVarCharWriter)
+            .put("CharType", SnowflakeClient::snowFlakeCharWriter)
+            .put("LongDecimalType", SnowflakeClient::snowFlakeDecimalWriter)
+            .put("ShortDecimalType", SnowflakeClient::snowFlakeDecimalWriter)
+            .buildOrThrow();
+
+    @Inject
+    public SnowflakeClient(
+            BaseJdbcConfig config,
+            ConnectionFactory connectionFactory,
+            QueryBuilder queryBuilder,
+            IdentifierMapping identifierMapping,
+            RemoteQueryModifier remoteQueryModifier)
+    {
+        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), identifierMapping, remoteQueryModifier, false);
+
+        JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+        ConnectorExpressionRewriter<ParameterizedExpression> connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
+                .addStandardRules(this::quoted)
+                .build();
+
+        this.aggregateFunctionRewriter = new AggregateFunctionRewriter<>(
+                connectorExpressionRewriter,
+                ImmutableSet.<AggregateFunctionRule<JdbcExpression, ParameterizedExpression>>builder()
+                        .add(new ImplementCountAll(bigintTypeHandle))
+                        .add(new ImplementCount(bigintTypeHandle))
+                        .add(new ImplementMinMax(false))
+                        .add(new ImplementSum(SnowflakeClient::toTypeHandle))
+                        .add(new ImplementAvgFloatingPoint())
+                        .add(new ImplementAvgDecimal())
+                        .build());
+    }
+
+    @Override
+    public void abortReadConnection(Connection connection, ResultSet resultSet)
+            throws SQLException
+    {
+        // Abort connection before closing. Without this, the Snowflake driver
+        // attempts to drain the connection by reading all the results.
+        connection.abort(directExecutor());
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        String jdbcTypeName = typeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new TrinoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+        jdbcTypeName = jdbcTypeName.toLowerCase(Locale.ENGLISH);
+        int type = typeHandle.getJdbcType();
```

Comment:
what about forced mapping to varchar? the code is accepting the config for it in the ctor but not using it.